### ISSUE TITLE
Fix imports changes more than is required

### ISSUE THREAD

**Issue by eamodio** [Microsoft/vscode#54969]
Issue Type: <b>Bug</b>

When renaming or moving a file and answering yes to fixing up the imports, causes unrelated imports to be changed.

For example, renaming a completely unrelated file, caused this import to change in a bunch of files.
```ts
import { /* */ } from “vscode-languageserver”
```
to
```ts
import { /* */ } from “vscode-languageserver/lib/main”
```

It seems that the fix up shouldn't change anything other than the imports that are required to change because of the rename/move operation.

At the same time, I don't know why it would change to that deeper import anyway -- which seems similar to https://github.com/Microsoft/vscode/issues/54841

//cc @mjbvz 

VS Code version: Code - Insiders 1.26.0-insider (befea2253626b1918352df04477cae83eee618ac, 2018-07-24T05:20:05.823Z)
OS version: Windows_NT x64 10.0.17713

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz (8 x 4008)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: disabled_software<br>video_decode: enabled<br>video_encode: enabled<br>vpx_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|31.93GB (18.19GB free)|
|Process Argv|C:\Users\Eric\AppData\Local\Programs\Microsoft VS Code Insiders\Code - Insiders.exe C:\Users\Eric\code\codestream\vscode-codestream\codestream.code-workspace|
|Screen Reader|no|
|VM|0%|

</details>

**Comment by mjbvz** [comment]
I can't repo this in a small example project using TS-3.0.1-insiders.20180723

Do you have an example project where this is happening? Please share the jsconfig/tsconfig and locations of the files being renamed at least

**Comment by eamodio** [comment]
I'm still using Typescript 2.92 -- is that why? If not, then I'll look into providing an example.

**Comment by tomfumb** [comment]
I have observed a similar issue in TypeScript 2.5.2, e.g. when renaming a file
`import { State } from '@progress/kendo-data-query';`
becomes
`import { State } from '@progress/kendo-data-query/dist/es/main';`
in files that do not even reference the renamed file.
My project builds fine but some of the import changes (especially with rxjs) result in ambiguous runtime errors that are very difficult to trace.
VSCode 1.25.1

**Comment by mjbvz** [comment]
Yes this is only fixed in TS 3.0+. Let me know if you still see it using a newer version
